src/main/java/hudson/tasks/junit/SuiteResult.java

```diff
@@ -126,22 +154,23 @@
         Document result = saxReader.read(xmlReport);
         Element root = result.getRootElement();
 
-        parseSuite(xmlReport,keepLongStdio,r,root);
+        parseSuite(xmlReport,keepLongStdio,r,root, runId, nodeId);
 
         return r;
     }
 
-    private static void parseSuite(File xmlReport, boolean keepLongStdio, List<SuiteResult> r, Element root) throws DocumentException, IOException {
+    private static void parseSuite(File xmlReport, boolean keepLongStdio, List<SuiteResult> r, Element root, String runId,
```

Comment:
Recommend making heavy use of CheckForNull in the runId and nodeId fields/parameters we've added.